src/modules/commander/Commander.cpp

```diff
@@ -1984,6 +1987,47 @@ void Commander::safetyButtonUpdate()
 	}
 }
 
+void Commander::throwLaunchUpdate()
+{
+	if (_param_com_throw_en.get()) {
+		if (_vehicle_local_position_sub.updated()) {
+			_vehicle_local_position_sub.copy(&_vehicle_local_position);
+		}
+
+		if (!isArmed() && _throw_launch_state != ThrowLaunchState::IDLE) {
+			mavlink_log_info(&_mavlink_log_pub, "The vehicle is DISARMED with throw launch enabled. Do NOT throw it.\t");
+			_throw_launch_state = ThrowLaunchState::IDLE;
+		}
+
+		if (_throw_launch_state == ThrowLaunchState::IDLE && isArmed()) {
+			mavlink_log_info(&_mavlink_log_pub, "The vehicle is ARMED with throw launch enabled. Throw the vehicle now.\t");
+			_throw_launch_state = ThrowLaunchState::ARMED;
+		}
+
+		float vehicle_speed_squared = (
+						      _vehicle_local_position.vx * _vehicle_local_position.vx +
+						      _vehicle_local_position.vy * _vehicle_local_position.vy +
+						      _vehicle_local_position.vz * _vehicle_local_position.vz
+					      );
+		float min_launch_speed = _param_com_throw_min_speed.get();
+
+		if (_throw_launch_state == ThrowLaunchState::ARMED &&
+		    vehicle_speed_squared >= min_launch_speed * min_launch_speed) {
```

Comment:
you can use `Vector3f(_vehicle_local_position.vx, _vehicle_local_position.vy, _vehicle_local_position.vz).longerThan(_param_com_throw_min_speed.get())`

`longerThan`also makes the comparison using the norm squared to avoid a sqrt

Reply:
Nice suggestion, together with the switch case looks better now!